tests/probes_test.go

```diff
@@ -61,19 +63,13 @@ var _ = Describe("[ref_id:1182]Probes", func() {
 
 		table.DescribeTable("should fail", func(readinessProbe *v12.Probe) {
 			By("Specifying a VMI with a readiness probe")
-			vmi := createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
+			vmi = createReadyCirrosVMIWithReadinessProbe(virtClient, readinessProbe)
 
-			Expect(tests.PodReady(tests.GetRunningPodByVirtualMachineInstance(vmi, tests.NamespaceTestDefault))).To(Equal(v1.ConditionFalse))
-			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
-			Expect(vmiReady(vmi)).To(Equal(v1.ConditionFalse))
+			// pod is not ready until our probe contacts the server
+			assertPodNotReady(virtClient, vmi)
 
 			By("Checking that the VMI and the pod will consistently stay in a not-ready state")
-			Consistently(func() v1.ConditionStatus {
-				vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &v13.GetOptions{})
-				Expect(err).ToNot(HaveOccurred())
-				return vmiReady(vmi)
-			}, 60, 1).Should(Equal(v1.ConditionFalse))
+			Consistently(isVMIReady).Should(Equal(false))
```

Comment:
I think you will miss here the case the condition is unknown.

Since `ConditionStatus` has more values than `true` and `false`. I think it is safer to check explicitly you get the condition you expect to get.

Reply:
I guess it is staring with `unknown` and then moves to `false` after several retries. I `ConditionStatus` and treating it as boolean my cover bugs.
But I won't insist on it.